azurerm/internal/services/sentinel/sentinel_data_connector.go

```diff
@@ -0,0 +1,86 @@
+package sentinel
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/securityinsight/mgmt/2019-01-01-preview/securityinsight"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataConnectorID(dc securityinsight.BasicDataConnector) *string {
```

Comment:
as mentioned in #10671 - I think we can remove this in favour of using the ID parser instead?